room.cpp

```diff
@@ -0,0 +1,53 @@
+#include "room.h"
+
+Room::Room(const std::string &name): name_{name} {}
+
+void Room::AddSensor(Sensor &sensor)
+{
+    sensors_.insert({sensor.GetName(), sensor});
+}
+
+void Room::DeleteSensor(const std::string &name)
+{
+    sensors_.erase(sensors_.find(name));
+}
+
+Sensor* Room::GetSensor(const std::string &name)
+{
+    return &sensors_.find(name)->second;
```

Comment:
Can you explain why you're using a pointer return type(not saying it's wrong, just want to see your reasoning).
Usually we see returns by ref, const ref or value.

Reply:
Honestly, I was in a hurry. I tried with ref the first time and something went wrong (I recreated that scenario now and I found out my mistake) and I thought to try with pointer because the efficency is the same at both from what I know